tests/test_scripts.py

```diff
@@ -132,3 +136,309 @@ def test_make_query(self):
+
+        with open(migration_file_info[1], 'w') as migration:
+            # Write content to the file.
+            migration.write(content)
+
+    def setup(self):
+        super(TestDatabaseMigrationScript, self).setup()
+        self.script = DatabaseMigrationScript(_db=self._db)
+
+        # This list holds any temporary files created during tests
+        # so they can be deleted during teardown().
+        self.migration_files = []
+        self._create_test_migrations()
+
+        stamp = datetime.datetime.strptime('20260810', '%Y%m%d')
+        self.timestamp = Timestamp(service=self.script.name, timestamp=stamp)
+        self._db.add(self.timestamp)
+
+    def teardown(self):
+        """Delete any created records, files and directories."""
+
+        test_timestamps = self._db.query(Timestamp).filter(
+            Timestamp.service.like('Test Database Migration Script - %')
+        )
+        for timestamp in test_timestamps.all():
+            self._db.delete(timestamp)
+
+        for fd, fpath in self.migration_files:
+            os.close(fd)
+            os.remove(fpath)
+            if fpath.endswith('.py'):
+                # Remove compiled files.
+                try:
+                    os.remove(fpath+'c')
+                except OSError:
+                    pass
+
+        for directory in [self.core_migration_dir, self.parent_migration_dir]:
+            if not os.listdir(directory):
+                os.rmdir(directory)
+
+        super(TestDatabaseMigrationScript, self).teardown()
+
+    def test_directories_by_priority(self):
+        core = os.path.split(os.path.split(__file__)[0])[0]
+        parent = os.path.split(core)[0]
+        expected_core = os.path.join(core, 'migration')
+        expected_parent = os.path.join(parent, 'migration')
+
+        eq_(
+            [expected_core, expected_parent],
+            self.script.directories_by_priority
+        )
+
+    def test_fetch_migration_files(self):
+        result = self.script.fetch_migration_files()
+        result_migrations, result_migrations_by_dir = result
+
+        for desc, migration_file in self.migration_files:
+            assert os.path.split(migration_file)[1] in result_migrations
+
+        def extract_filenames(core=True):
+            pathnames = [pathname for desc, pathname in self.migration_files]
+            if core:
+                pathnames = [p for p in pathnames if 'core' in p]
+            else:
+                pathnames = [p for p in pathnames if 'core' not in p]
+
+            return [os.path.split(p)[1] for p in pathnames]
+
+        # Ensure that all the expected migrations from CORE are included in
+        # the 'core' directory array in migrations_by_directory.
+        core_migration_files = extract_filenames()
+
+        # At some point, August 10th, 2026 will come and go and a migration
+        # may be added and this unit test will start to fail right below.
+        # At that point, move all dates to the 2030s.
```

Comment:
It seems like `directories_by_priority` is the problem here. Can you create a subclass of `DatabaseMigrationScript` which overrides `directories_by_priority` to return the prepopulated temp directories?